Practica1_TSP/TSP_funcional/src/heuristicas/HeuristicaVMC.java

```diff
@@ -0,0 +1,116 @@
+   /**
+    * referencia al problema a resolver
+    */
+   private final MapaTSP mapa;
+
+   /**
+    * constructor de la clase
+    * @param mapa mapa del problema
+    */
+   public HeuristicaVMC(MapaTSP mapa){
+      // se asigna el mapa
+      this.mapa = mapa;
+   }
+
+   /**
+    * metodo de resolucion con heuristica del vecino
+    * mas cercano
+    * @return la ruta optima
+    */
+   public Ruta resolver(){
+      System.out.println("Interfaz heuristicas.HeuristicaVMC");
+      System.out.println("resolucion mediante vecino mas cercano");
+
+      Ruta optima = new Ruta();
+      ArrayList<Ruta> rutas = new ArrayList<>();
+
+      // Para cada ciudad i
+      this.mapa.obtenerPuntos().forEach(ciudad -> {
+         Ruta resultado = new Ruta();
+
+         // Añadir la primera ciudad al recorrido
+         double coste = resultado.calcularLongitud() == 0 ?
+                 0.0 : this.mapa.calcularDistancia(resultado.obtenerUltimo(),ciudad);
+         resultado.agregar(ciudad, coste);
+
+         // Completar la ruta
+         this.completarRuta(resultado);
+
+         // Se agrega la ruta nueva a la coleccion de rutas
+         rutas.add(resultado);
+      });
+
+      // Una vez aqui el array de rutas tiene una ruta de vecino mas cercano por cada ciudad del mapa
+      // Obtener de todas las rutas la ruta optima, es decir la de menor coste
+      optima = this.obtenerRutaMenorCoste(rutas);
+
+      System.out.println("RESULTADO VCM:");
+      System.out.println(optima.toString());
+
+      // se devuelve la ruta de minimo coste
+      return optima;
+   }
+
+   /**
+    * Metodo recursivo para completar la ruta con las ciudades del mapa
+    * @param ruta a completar
+    */
+   private void completarRuta(Ruta ruta){
+      // Si la ruta no está completa
+      if(ruta.calcularLongitud() < this.mapa.obtenerDimension() ){
+         // Obtener el punto mas cercano al anterior (que no haya sido visitada)
+         Punto masCercana = this.puntoMasCercanoValido(ruta);
+
+         // Añadirlo al recorrido
+         double coste = ruta.calcularLongitud() == 0 ?
+                 0.0 : this.mapa.calcularDistancia(ruta.obtenerUltimo(),masCercana);
+         ruta.agregar(masCercana, coste);
+
+         // Recursividad: seguir completando la ruta
+         this.completarRuta(ruta);
+      }
+      else {
+         // Si la ruta ya esta completa
+         // Volvemos a añadir la primera ciudad para finalizar el recorrido
+         double coste = ruta.calcularLongitud() == 0 ?
+                 0.0 : this.mapa.calcularDistancia(ruta.obtenerUltimo(),ruta.obtenerPunto(0));
+         ruta.agregar(ruta.obtenerPunto(0), coste);
+      }
+   }
+
+   /**
+    * Metodo para obtener el siguiente punto del mapa más cercano al último punto de la ruta y que no haya sido visitado
+    * @param ruta ruta actual
+    * @return devuelve el punto mas cercano no visitado
+    */
+   private Punto puntoMasCercanoValido(Ruta ruta) {
```

Comment:
So cool this way 👌 

Don't you think MapaTSP class is more suitable for this method?